garak/generators/huggingface.py

```diff
@@ -468,6 +486,12 @@ def _load_client(self):
                 self.name, padding_side="left"
             )
 
+        # test tokenizer for `apply_chat_template` support
+        self.use_chat = (
+            hasattr(self.tokenizer, "chat_template")
+            and self.tokenizer.chat_template is not None
+        )
+
```

Comment:
This logic is in a few places, is it worth factoring up to `HFCompatible`?

Reply:
the object format is slightly different `self.tokenizer` vs `self.generator.tokenizer` there is an abstraction possible here for `_supports_chat(tokenizer)` though not sold on the value yet:

``` python
def _supports_chat(self, tokenizer) -> bool:
    return hasattr(tokenizer, "chat_template") and tokenizer.chat_template is not None
```

I would like to defer this until I see a third usage or usage outside this module.